cores/esp8266/Esp.h

```diff
@@ -131,6 +133,13 @@ class EspClass {
         bool eraseConfig();
 
         inline uint32_t getCycleCount();
+
+        /**
+         * Be very careful in your callback.  Don't call anything that could
+         *  yield (so pretty much don't call Arduino code).  And god help you if
+         *  it crashes.
+         */
+        void installCustomCrashCallback( CrashCallback );
```

Comment:
My suggestion is to install custom crash handler by overriding some weak
function, which defaults to no-op.




On Thu, Nov 12, 2015 at 9:56 PM, Carl Pacey <notifications@github.com>
wrote:

> In cores/esp8266/Esp.h
> <https://github.com/esp8266/Arduino/pull/1012#discussion_r44698297>:
>
> > @@ -131,6 +133,13 @@ class EspClass {
> >          bool eraseConfig();
> >
> >          inline uint32_t getCycleCount();
> > +
> > +        /**
> > +         * Be very careful in your callback.  Don't call anything that could
> > +         *  yield (so pretty much don't call Arduino code).  And god help you if
> > +         *  it crashes.
> > +         */
> > +        void installCustomCrashCallback( CrashCallback );
>
> This was the best place I could find to allow installing the callback.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/esp8266/Arduino/pull/1012/files#r44698297>.
>